cosmo_tester/resources/blueprints/active-directory/scripts/wait_for_active_directory_port.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+HOST_IP=$public_ip
+PORT=$port
+TIMEOUT_MIN="$(($timeout_minutes * 60))"
+function wait_for_port
+{
+    /bin/netcat -z -w$3 $1 $2
+    if [ $? -ne 0 ] && $fail_on_timeout; then
```

Comment:
1. what if fail_on_timeout is set to False and the netcat command fails not because of the timeout?
it should probably be something like:
if return_code is not 0:
  if fail_on_timeout:
    echo and exit 1
  else:
    echo and do something else
2. the error message writes "minutes" but it's actually in seconds